optimum/exporters/onnx/base.py

```diff
@@ -587,13 +588,26 @@ def outputs(self) -> Mapping[str, Mapping[int, str]]:
         common_outputs = super(OnnxConfigWithPast, self).outputs
         # Renaming the outputs axes properly.
         for name, axes_names in common_outputs.items():
-            sequence_name = "encoder_sequence_length" if "encoder" in name else "decoder_sequence_length"
-            for axis_idx, name in axes_names.items():
-                if "sequence" in name:
-                    axes_names[axis_idx] = sequence_name
-                # We reset the value as the order in common_outputs (OrderedDict) is lost otherwise
+            if self._behavior is ConfigBehavior.ENCODER:
+                sequence_name = "encoder_sequence_length"
+            else:
+                if "encoder" in name:
+                    sequence_name = "encoder_sequence_length"
                 else:
-                    axes_names[axis_idx] = name
+                    sequence_name = "decoder_sequence_length"
+
+            new_axes_names = {}
+            for axis_idx, axis_name in axes_names.items():
+                if "sequence" in axis_name:
+                    if not self.use_past_in_inputs:
+                        new_axes_names[axis_idx] = sequence_name
+                    else:
+                        # Trick to force it since ONNX sometimes infer a dynamic axis where it's not.
+                        new_axes_names[axis_idx] = "1"
```

Comment:
Does this happen also when we do not put the axis_id in dict?

Reply:
Yes, basically I do that because ONNX infers that the axis is dynamic but it is not. I will look into this problem, which is not limited to this case, on another PR.